libsql_client/hrana/conn.py

```diff
@@ -53,6 +55,11 @@ def __init__(self, session: aiohttp.ClientSession, url: str, auth_token: Optiona
 
         self._send({"type": "hello", "jwt": auth_token})
 
+    async def wait_connected(self) -> None:
+        await self._finished_handshake.wait()
+        if self.exception:
+            raise self.exception
```

Comment:
This one I need in dbapi2, otherwise we can't work as expected, where you should be getting connection opening errors.

It's harmless to not use it, it won't block anything.

And in dbapi2 I can quickly get the errors, I tried polling (busy wait) but it was harming CPU usage.